### ISSUE TITLE
Routing mode uses incorrect camera (sometimes)

### ISSUE THREAD

**Issue by ecgreb** [mapzen/eraser-map#602]

**Comment by mjcunningham** [comment]
If you go in and out of routing mode a few times you can occasionally get it to load the isometric camera instead of the perspective camera. If you go in and out a few more times it'll fix itself or crash.

**Comment by mjcunningham** [comment]
This still exists in master 1609

**Comment by ecgreb** [comment]
I'm no longer able to reproduce on my device.

@blair1618 @tallytalwar do you see anything wrong with how we are [updating the scene file](https://github.com/mapzen/eraser-map/blob/master/app/src/main/kotlin/com/mapzen/erasermap/controller/MainActivity.kt#L1145)? `MapzenMap#queueSceneUpdate(...)` and `MapzenMap#applySceneUpdates()` just wrap the `MapController` methods of the same name.

**Comment by blair1618** [comment]
Doesn't look obviously wrong. What's the reason for that `if` statement wrapping the scene update?

**Comment by ecgreb** [comment]
@blair1618 nice catch! I looked at this code so many times I wasn't even seeing the `if` anymore 😛 

If you pan the map in routing mode to show the resume button, then exit routing, then start nav again the camera does not update.

This check was added as part of https://github.com/mapzen/eraser-map/pull/494 to prevent resetting the map position when the user is manipulating the map and a reroute occurs. I'm not sure this check is needed though as we will want to remain in perspective camera regardless.
